src/huggingface_hub/hf_file_system.py

```diff
@@ -620,7 +620,7 @@ def _fetch_range(self, start: int, end: int) -> bytes:
             repo_type=self.resolved_path.repo_type,
             endpoint=self.fs.endpoint,
         )
-        r = http_backoff("GET", url, headers=headers)
+        r = http_backoff("GET", url, headers=headers, retry_on_status_codes=502)
```

Comment:
Let's keep 503 (the default one) and add 502 (this PR) and 504 as well (also come up sometimes)

```suggestion
        r = http_backoff("GET", url, headers=headers, retry_on_status_codes=(502, 503, 504))
```